tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -101,14 +108,45 @@ def read(self, filename, binary_mode=False, size=None, offset=None):
             Subset of the contents of the file as a string or bytes.
         """
         mode = "rb" if binary_mode else "r"
-        with io.open(filename, mode) as f:
+        encoding = None if binary_mode else "utf8"
+        with io.open(filename, mode, encoding=encoding) as f:
             if offset is not None:
                 f.seek(offset)
             if size is not None:
                 return f.read(size)
             else:
                 return f.read()
 
+    def write(self, filename, file_content, binary_mode=False):
+        """Writes string file contents to a file.
+
+        Args:
+            filename: string, a path
+            file_content: string, the contents
+            binary_mode: bool, write as binary if True, otherwise text
+        """
+        self._write(filename, file_content, "wb" if binary_mode else "w")
+
+    def append(self, filename, file_content, binary_mode=False):
+        """Append string file contents to a file.
+
+        Args:
+            filename: string, a path
+            file_content: string, the contents to append
+            binary_mode: bool, write as binary if True, otherwise text
+        """
+        self._write(filename, file_content, "ab" if binary_mode else "a")
+
+    def _write(self, filename, file_content, mode):
+        if "b" in mode:
+            encoding = None
+            file_content = compat.as_bytes(file_content)
```

Comment:
I'd still prefer to avoid doing the auto-conversions here between string and bytes - I think it's just better to start out with the stricter behavior and if that proves too onerous we could relax it later, whereas it's backwards incompatible to go the other direction.  For torch.utils.tensorboard I think all it requires is being clear about whether you're writing text or bytes at a given callsite which should be straightforward - and I'd expect that generally you're only dealing with bytes, right?  Certainly for the EventFileWriter it's only bytes.

Reply:
Let me see what I can do.